src/opm/input/eclipse/EclipseState/SimulationConfig/ThresholdPressure.cpp

```diff
@@ -127,6 +128,38 @@ namespace Opm {
         }
     }
 
+    void ThresholdPressure::readFaults(const Deck& deck,
+                                       const FaultCollection& faults)
+    {
+        if (!DeckSection::hasGRID(deck) || !DeckSection::hasSOLUTION(deck))
+            return;
+
+        GRIDSection gridSection( deck );
+
+        const bool thpresftKeyword  = gridSection.hasKeyword<ParserKeywords::THPRESFT>();
+        if (!thpresftKeyword)
+            return;
+
+        // extract the multipliers from the deck keyword
+        m_thresholdFaultTable.resize(faults.size(), -1.0);
+        SOLUTIONSection solutionSection(deck);
+        const auto& thpresft = solutionSection.get<ParserKeywords::THPRESFT>().back();
+        for (size_t recordIdx = 0; recordIdx < thpresft.size(); ++ recordIdx) {
+            const DeckRecord& record = thpresft.getRecord(recordIdx);
+
+            const std::string& faultName = record.getItem("FAULT_NAME").getTrimmedString(0);
+            double thpresValue = record.getItem("VALUE").getSIDouble(0);
+
+            for (size_t faultIdx = 0; faultIdx < faults.size(); faultIdx++) {
+                auto& fault = faults.getFault(faultIdx);
+                if (fault.getName() != faultName)
+                    continue;
```

Comment:
How is this usually resolved?

Reply:
> How is [templated strings] usually resolved?

The least invasive approach is probably to use our home-grown shell wildcard matching function [`shmatch()`](https://github.com/OPM/opm-common/blob/0624a04613d9b011d5a9b81d442cd198d71df562/opm/common/utility/shmatch.hpp#L33):
```C++
if (! shmatch(faultName, fault.getName()) {
    continue;
}
```
Ideally the `FaultCollection` would have a way to do this too, but apparently it does not, at least not right now.

Reply:
So this will run a regex, but it won't give back a list of matches. Is that sufficient?

Reply:
never mind, we loop around it of course.

Reply:
> So this will run a regex, but it won't give back a list of matches. Is that sufficient?

We're already N-squared here, we loop the complete list of faults for every `THPRESFT`  record, so we only need to know if `fault.getName()` matches the `faultName` template.  We'd need a radically different strategy if having a list of matches would be useful.